### PR TITLE
fix text not being updated if textProvider is set after animation

### PR DESCRIPTION
**Pull request by RolandasRazma**
fix text not being updated if textProvider is set after animation. This doesn't fix case where text change - you will need to set new (or same) provider again

### CODE REVIEW

**Review by buba447 — APPROVED** [review]

**Comment by calda** [comment]
@RolandasRazma, this seems like a reasonable change. We can probably merge this after rebasing / resolving merge conflicts.

**Comment by calda** [comment]
Closing since there are unresolved merge conflicts. Feel free to re-open once the merge conflicts are resolved.

**Pull request closed without merging**
